src/ig/IGExporter.ts

```diff
@@ -1479,6 +1589,69 @@ export class IGExporter {
       }
     }
   }
+
+  // Add an extension for the R5 property IG.definition.page.source[x] if provided in configuration
+  addNameUrlAndPageSourceExtensionForR4(
+    page: ImplementationGuideDefinitionPage,
+    configPages: ImplementationGuideDefinitionPage[]
+  ): void {
+    const sourceExtensionUrl =
+      'http://hl7.org/fhir/5.0/StructureDefinition/extension-ImplementationGuide.definition.page.source';
+    const nameExtensionUrl =
+      'http://hl7.org/fhir/5.0/StructureDefinition/extension-ImplementationGuide.definition.page.name';
+    const configPage = configPages?.find(
+      p =>
+        (p.nameUrl || p.name).substring(0, (p.nameUrl || p.name).lastIndexOf('.')) ===
+        page.nameUrl.substring(0, page.nameUrl.lastIndexOf('.'))
+    );
+    if (configPage) {
+      if (configPage.sourceUrl) {
+        // If nameUrl and sourceUrl do not match, add the sourceUrl value to an extension
+        if (configPage.nameUrl !== configPage.sourceUrl) {
+          page.extension = (page.extension ?? []).concat({
+            url: sourceExtensionUrl,
+            valueUrl: configPage.sourceUrl
+          });
+        }
+
+        if (configPage.name) {
+          // Add extension for name if present
+          page.extension = (page.extension ?? []).concat({
+            url: nameExtensionUrl,
+            valueUrl: configPage.name
+          });
+        }
+      } else {
+        if (configPage.name) {
+          // If nameUrl and name do not match, add the name value to an extension
+          if (configPage.nameUrl !== configPage.name) {
+            page.extension = (page.extension ?? []).concat({
+              url: nameExtensionUrl,
+              valueUrl: configPage.name
+            });
+          }
+        }
+        // Once nameUrl is set, assign a configured source[x] to an extension
+        if (configPage.sourceString) {
```

Comment:
Don't we only need to do this if `configPage.sourceString and configPage.nameUrl !== configPage.sourceString`?

Reply:
Honestly, I forget everything we decided that day in the lab, so I could be wrong here...

Reply:
My apologies. I was thinking of `sourceUrl`.  You're right. `sourceString` is different -- so leave this as-is.